src/dotnet/Chat.Service/ChatsMigrationBackend.cs

```diff
@@ -0,0 +1,449 @@
+                            newAuthorSid),
+                    cancellationToken)
+                .ConfigureAwait(false);
+#pragma warning restore CA1307
+            Log.LogInformation("Updated AuthorId for {Count} reaction records with old author id '{AuthorId}'."+
+                " New author id is '{NewAuthorId}'", updateCount, authorSid, newAuthorSid);
+        }
+
+#pragma warning disable CA1307
+        updateCount = await dbContext.Reactions
+            .Where(c => c.Id.StartsWith(chatSid))
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.EntryId,
+                        c => c.EntryId.Replace(chatSid, newChatSid))
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated EntryId and Id for {Count} reaction records", updateCount);
+    }
+
+    private async Task UpdateReactionSummaries(
+        ChatDbContext dbContext,
+        string chatSid,
+        string newChatSid,
+        IReadOnlyCollection<MigratedAuthor> migratedAuthors,
+        CancellationToken cancellationToken)
+    {
+        #pragma warning disable CA1307
+        var updateCount = await dbContext.ReactionSummaries
+            .Where(c => c.Id.StartsWith(chatSid))
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.EntryId,
+                        c => c.EntryId.Replace(chatSid, newChatSid))
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated EntryId and Id for {Count} reaction summary records", updateCount);
+
+        updateCount = 0;
+        foreach (var dbSummary in dbContext.ReactionSummaries.Where(c => c.Id.StartsWith(chatSid))) {
+            var summary = dbSummary.ToModel();
+            var authorIds = summary.FirstAuthorIds;
+            var newAuthorIds = ImmutableList<AuthorId>.Empty;
+            foreach (var authorId in authorIds) {
+                var newAuthorId = migratedAuthors.First(c => OrdinalEquals(c.OriginalAuthor.Id.Value, authorId.Value)).NewId;
+                newAuthorIds = newAuthorIds.Add(newAuthorId);
+            }
+            summary = summary with { FirstAuthorIds = newAuthorIds };
+            dbSummary.UpdateFrom(summary);
+            updateCount++;
+        }
+
+        await dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+
+        Log.LogInformation("Updated AuthorIds for {Count} reaction summary records", updateCount);
+    }
+
+    private static void AssertOneUpdated(int updateCount)
+        => AssertXUpdated(updateCount, 1);
+
+    private static void AssertXUpdated(int updateCount, int expectedUpdateCount)
```

Comment:
Rename to `AssertUpdated`; also, I'd rather use `N` instead of `X` in similar cases: `N` is more associated with a number, though `X` can be anything.

Reply:
Fixed